src/utils/links.ts

```diff
@@ -15,8 +27,52 @@ export function makeLinks(itemQuery: string, instanceSettings: DataSourceInstanc
   ];
 }
 
+export async function addTraceToLogsLinks(frame: DataFrame, region?: string, datasourceUid?: string) {
+  if (!datasourceUid) {
+    return;
+  }
+  try {
+    const spanField = frame.fields.find((f) => f.name === 'spanID')!;
+    const requestIdField = frame.fields.find((f) => f.name === '__request_id')!;
+    const link = await makeTraceToLogsLink(datasourceUid, region, {
+      hasRequestId: Boolean(requestIdField.values.get(0)),
+    });
+    spanField.config.links = [link];
+  } catch (error) {
+    // There are some things that can go wrong like datasourceUID not existing anymore etc. Does not seem useful to
+    // error the whole query in that case so we will just skip the links.
+    console.error(error);
+  }
+}
+
+async function makeTraceToLogsLink(
+  datasourceUid: string,
+  region = 'default',
+  options: { hasRequestId: boolean }
+): Promise<DataLink> {
+  const logsDS = await getDataSourceSrv().get(datasourceUid);
+  const filter = options.hasRequestId
+    ? 'filter @requestId = "${__data.fields.__request_id}" or @message like "${__data.fields.traceID}"'
```

Comment:
This seems to be a filter that also x-ray uses when looking for logs from a trace in their UI.